### PR TITLE
[SPARK-29392][CORE][SQL][FOLLOWUP] Avoid deprecated (in 2.13) Symbol syntax 'foo in favor of simpler expression, where it generated deprecation warnings

### PR DESCRIPTION
**Pull request by srowen**
### What changes were proposed in this pull request?

Where it generates a deprecation warning in Scala 2.13, replace Symbol shorthand syntax `'foo` with an equivalent.

### Why are the changes needed?

Symbol syntax `'foo` is deprecated in Scala 2.13. The lines changed below otherwise generate about 440 warnings when building for 2.13.

The previous PR directly replaced many usages with `Symbol("foo")`. But it's also used to specify Columns via implicit conversion (`.select('foo)`) or even where simple Strings are used (`.as('foo)`), as it's kind of an abstraction for interned Strings.

While I find this syntax confusing and would like to deprecate it, here I just replaced it where it generates a build warning (not sure why all occurrences don't): `$"foo"` or just `"foo"`.

### Does this PR introduce any user-facing change?

Should not change behavior.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by Gschiavon** [comment]
@srowen this means the `'foo` sintax will be not longer available?

**Comment by srowen** [comment]
Oh no, that syntax is still present in 2.13, it's just deprecated. The implicit conversions in Spark, to things like Column, are still there too. I just changed the instances that generate a warning in 2.13 (and I don't know why many do not?) to explicitly use a Column or String.

File: sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala [hunk 1]
```diff
@@ -39,7 +39,7 @@ import org.apache.spark.sql.types._
  * OrcFilterSuite and HiveOrcFilterSuite is logically duplicated to provide the same test coverage.
  * The difference are the packages containing 'Predicate' and 'SearchArgument' classes.
  * - OrcFilterSuite uses 'org.apache.orc.storage.ql.io.sarg' package.
- * - HiveOrcFilterSuite uses 'org.apache.hadoop.hive.ql.io.sarg' package.
+ * - HiveOrcFilterSuite uses $"org".apache.hadoop.hive.ql.io.sarg' package.
```
Comment (dongjoon-hyun) [inline sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala]: Shall we revert this line?

**Comment by dongjoon-hyun** [comment]
Also, please fix the following file together for `Hive-2.3` (default) profile.
- sql/core/v2.3/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala [hunk 2]
```diff
@@ -65,7 +65,8 @@ class PruneFileSourcePartitionsSuite extends QueryTest with SQLTestUtils with Te
           options = Map.empty)(sparkSession = spark)
 
         val logicalRelation = LogicalRelation(relation, tableMeta)
-        val query = Project(Seq('i, 'p), Filter('p === 1, logicalRelation)).analyze
+        val query = Project(Seq(Symbol("i"), Symbol("p")),
+          Filter(Symbol("p") === 1, logicalRelation)).analyze
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala]: nit.
```scala
-        val query = Project(Seq(Symbol("i"), Symbol("p")),
-          Filter(Symbol("p") === 1, logicalRelation)).analyze
+        val query = Project(Seq($"i", $"p"), Filter($"p" === 1, logicalRelation)).analyze
```

File: sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala (hunk 1 above)
Comment (srowen) [reply sql/core/v1.2/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala]: Oops, my search and replace was too aggressive

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (with only the above 3 comments).

**Review by HyukjinKwon — APPROVED** [review]
Looks good to me too otherwise.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala (hunk 2 above)
Comment (srowen) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala]: Here, I honestly wasn't sure whether I should swap out a Symbol for a Column reference. Where the test code is directly invoking catalyst primitives, I generally left it with a more conservative change, to just swap in Symbol. I imagine that a Column reference here could be fine too, but wanted to limit the scope of the change.

Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala]: Got it. No problem~

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**